hazelcast-integration/spring-jcache/src/main/java/com/hazelcast/spring/jcache/SpringJCacheJavaConfig.java

```diff
@@ -0,0 +1,24 @@
+package com.hazelcast.spring.jcache;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.core.Hazelcast;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import static com.hazelcast.spring.jcache.DummyBeanTest.*;
+
+/**
+ * Example of Spring Caching with JCache
+ */
+public class SpringJCacheJavaConfig {
+    public static void main(String[] args) {
+        System.setProperty("hazelcast.jcache.provider.type", "client");
```

Comment:
Well, it's true if hazecast.jar and hazelcast-client.jar in a classpath
client caching provider will be used. In this case explicit configuration
maybe verbose.
As I understand it's nothing to do with SpringHazelcastCachingProvider.
Just how JCache SPI works
On Thu, Nov 19, 2015 at 04:40 Mustafa İman <notifications@github.com> wrote:

> In
> hazelcast-integration/spring-jcache/src/main/java/com/hazelcast/spring/jcache/SpringJCacheJavaConfig.java
> <https://github.com/hazelcast/hazelcast-code-samples/pull/83#discussion_r45318704>
> :
>
> > @@ -0,0 +1,24 @@
> > +package com.hazelcast.spring.jcache;
> > +
> > +import com.hazelcast.client.HazelcastClient;
> > +import com.hazelcast.core.Hazelcast;
> > +import org.springframework.context.ApplicationContext;
> > +import org.springframework.context.annotation.AnnotationConfigApplicationContext;
> > +
> > +import static com.hazelcast.spring.jcache.DummyBeanTest.*;
> > +
> > +/**
> > + * Example of Spring Caching with JCache
> > + */
> > +public class SpringJCacheJavaConfig {
> > +    public static void main(String[] args) {
> > +        System.setProperty("hazelcast.jcache.provider.type", "client");
>
> SpringHazelcastCachingProvider handles this. It automatically detects if
> the instance is client or server and creates required type of caching
> provider.
>
> I am not even sure if setting the property is neccessary for annotation
> based configuration. Let me check it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hazelcast/hazelcast-code-samples/pull/83/files#r45318704>
> .
>
-- 
*Viktor Gamov*
*Senior Solution Architect*
Hazelcast | Open source in-memory data grid
275 7th Ave, New York, NY 10001 USA
@gamussa
+1 (201) 355-6226